lib/query.js

```diff
@@ -4429,10 +4521,12 @@ Query.prototype.exec = async function exec(op) {
       str = str.slice(0, 60) + '...';
     }
     const err = new MongooseError('Query was already executed: ' + str);
-    err.originalStack = this._executionStack;
+    if (!this.model.base.options.skipOriginalStackTraces) {
+      err.originalStack = this._executionStack;
+    }
     throw err;
   } else {
-    this._executionStack = new Error().stack;
+    this._executionStack = this.model.base.options.skipOriginalStackTraces ? true : new Error().stack;
```

Comment:
Something about this line doesn't feel right to me. In one scenario we're assigning a boolean, in another scenario we're assigning a string. Is this right?

Reply:
Yes it is expected unfortunately. The plan is to rename `_executionStack` and make it a boolean in 9.0. But for now this is a workaround to let people avoid `new Error().stack`, which can be expensive for apps that use source maps.